packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/repo/resource/Categories.java

```diff
@@ -0,0 +1,346 @@
+            return this;
+        }
+
+        @Override
+        public CategoryCreator category(String name)
+        {
+            return this.withName(name);
+        }
+
+        @Override
+        public CategoryCreator randomCategory()
+        {
+            return this.withRandomName();
+        }
+
+        @Override
+        public CategoryCreator randomCategory(String nameSuffix)
+        {
+            return this.withRandomName(nameSuffix);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator categories(String... names)
+        {
+            return new SerialCategoriesCreator(restClient, categoriesCache).withNames(names).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator randomCategories(String... nameSuffixes)
+        {
+            return new SerialCategoriesCreator(restClient, categoriesCache).withRandomNames(nameSuffixes).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator randomCategories(int quantity)
+        {
+            return new SerialCategoriesCreator(restClient, categoriesCache).withRandomNames(quantity).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator nestedCategories(String... names)
+        {
+            return new NestedCategoriesCreator(restClient, categoriesCache).withNames(names).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator nestedRandomCategories(String... nameSuffixes)
+        {
+            return new NestedCategoriesCreator(restClient, categoriesCache).withRandomNames(nameSuffixes).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public MultiCreator.CategoriesCreator nestedRandomCategories(int depth)
+        {
+            return new NestedCategoriesCreator(restClient, categoriesCache).withRandomNames(depth).underCategory(parent).asUser(user);
+        }
+
+        @Override
+        public CategoryCreator withRandomName()
+        {
+            return withRandomName(EMPTY);
+        }
+
+        @Override
+        public CategoryCreator withRandomName(String nameSuffix)
+        {
+            withAlias(nameSuffix);
+            return super.withRandomName(CATEGORY_NAME_PREFIX + nameSuffix + "_");
+        }
+        @Override
+        public CategoryCreator underCategory(RestCategoryModel parent)
+        {
+            this.parent = parent;
+            return this;
+        }
+
+        @Override
+        public RestCategoryModel create()
+        {
+            RestCategoryModel category = buildCategoryRestRequest(restClient, user, parent)
+                .createSingleCategory(RestCategoryModel.builder().name(name).create());
+
+            categoriesCache.put(alias, category);
+
+            return category;
+        }
+    }
+
+    public static class SerialCategoriesCreator
```

Comment:
There are a lot of classes in this file - it's a bit tricky to see what's here. I think it might be better to split them out into files of their own.

Reply:
I wanted to have all in one place, but you are probably right - this will clear a bit the code